Add Gitlab CI Pipeline docs to Port (#261)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/sentry.md

---
sidebar_position: 6
description: Ingest Sentry comments into your catalog
---

import SentryCommentsBlueprint from "./resources/sentry/\_example_sentry_comments_blueprint.mdx";
import SentryCommentsConfiguration from "./resources/sentry/\_example_sentry_comment_webhook_configuration.mdx"

# Sentry

In this example you are going to create a webhook integration between [Sentry](https://sentry.io) and Port, which will ingest comment entities.

## Prerequisites

Create the following blueprint definitions and webhook configuration:

<details>

<summary>Sentry comment blueprint</summary>
<SentryCommentsBlueprint/>

</details>

<details>

<summary>Sentry comment webhook configuration</summary>
<SentryCommentsConfiguration/>

</details>

:::tip

> We have left out the `secret` field from the security object in the webhook configuration. This is because we only get this value after creating the webhook integration. That said, follow the guide on [configuring webhook endpoint](../webhook.md#configuring-webhook-endpoints) to get your webhook URL and proceed to the below section.

Reviewer: ```suggestion
We have left out the `secret` field from the security object in the webhook configuration because the secret value is generated by Sentry when creating the webhook.

So when following this example, please first create the webhook configuration in Port. Use the webhook URL from the response and create the webhook in Sentry.

After getting the secret from Sentry, you can go back to Port and update the [webhook configuration](../webhook.md#configuring-webhook-endpoints) with the secret.
```

:::

## Create the Sentry webhook

1. Log in to Sentry with your organization's credentials;
2. Click the gear icon (Setting) at the left sidebar of the page;
3. Choose **Developer Settings**;
4. At the upper corner of this page, click on **Create New Integration**;

> 5. Sentry provides two types of integrations: Internal and Public. For this purpose of this guide, choose **Internal Integration** and click on the **Next** button

Reviewer: ```suggestion
5. Sentry provides two types of integrations: Internal and Public. For the purpose of this guide, choose **Internal Integration** and click on the **Next** button
```

6. Input the following details:
   1. `Name` - use a meaningful name such as Port Webhook;
   2. `Webhook URL` - enter the value of the `url` key you received after creating the webhook configuration;
   3. `Overview` - enter a description for the webhook;

>    4. `Permissions` - Grant your webhook the appropriate permissions based on the event you want to report;

Reviewer: ```suggestion
   4. `Permissions` - Grant your webhook **Read** permissions for the **Issue & Event** category;
```

   5. `Webhooks` - Under this section, enable the comments checkbox to allow Sentry to report comment events to Port;
7. Click **Save Changes** at the bottom of the page.

> ## Updating Webhook Configuration with Secret

Reviewer: Remove this section (but keep the summary paragraph at the end)

Add a :::tip block saying:
Now that the webhook is created, you can take the secret value generated by Sentry and use it to update the `security` object in your Port webhook configuration
